transport/processing/processing_component.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+import sys
+import yaml
+
+import gci.componentmodel as cm
+import processing.config as config
+import processing.processing as processing
+
+import ci.util
+import container
+import container.registry
+import product.v2
+
+
+def parse_component_descriptor(path: str):
+    with open(path) as desc_file:
+        return cm.ComponentDescriptor.from_dict(yaml.safe_load(desc_file))
+
+
+def _download_descriptor(name: str, version: str, ctx_base_url: str):
+    try:
+        return product.v2.download_component_descriptor_v2(
+                component_name=name,
+                component_version=version,
+                ctx_repo_base_url=ctx_base_url,
+        )
+    except container.registry.OciImageNotFoundException as err_not_found:
+        ci.util.error(err_not_found)
+        sys.exit(1)
+
+
+def _resolve_cd_from_references(component_descriptor: cm.ComponentDescriptor):
+    ctx_base_url = component_descriptor.component.repositoryContexts[0].baseUrl
+
+    def __enumerate_cd_from_references(
```

Comment:
nit: why name it w/ dunder-prefix? it is only visible locally, anyways

Reply:
no reason ;) fixed